sdk/core/System.ClientModel/src/Pipeline/ClientPipeline.cs

```diff
@@ -130,7 +140,14 @@ public static ClientPipeline Create(
         int perCallIndex = index;
 
         // Add retry policy.
```

Comment:
Would it be possible simplify the following logic to a single call to an internal API on an options/other object?